packages/l/libvorbis/xmake.lua

```diff
@@ -10,13 +10,67 @@ package("libvorbis")
 
     add_versions("1.3.7", "0e982409a9c3fc82ee06e08205b1355e5c6aa4c36bca58146ef399621b0ce5ab")
 
+    add_configs("vorbisenc",  {description = "Includes vorbisenc", default = true, type = "boolean"})
+    add_configs("vorbisfile", {description = "Includes vorbisfile", default = true, type = "boolean"})
+
     add_deps("cmake", "libogg")
 
+    on_fetch(function (package, opt)
+        if opt.system then
+            local vorbis = package:find_package("vorbis", opt)
+            if not vorbis then
+                return
+            end
+            local result = table.copy(vorbis)
+
+            if package:config("vorbisenc") then
+                local vorbisenc = package:find_package("vorbisenc", opt)
+                if not vorbisenc then
+                    return
+                end
+
+                result.includedirs = table.join(vorbisenc.sysincludedirs or vorbisenc.includedirs, result.includedirs)
```

Comment:
we should join result.includedir first. because vorbisenv.includedirs may be nil, it will break table.join and result.incluedirs will be missing 

```lua
result.includedirs = table.join(result.includedirs, vorbisenc.sysincludedirs or vorbisenc.includedirs)
```

```console
$ xmake l find_package libvorbisfile
{
  linkdirs = {
    "/usr/lib/x86_64-linux-gnu"
  },
  links = {
    "vorbisfile"
  }
}
```

Reply:
I reworked on_fetch based on what brotli does, it will probably be easier if you introduce a function to merge pkg infos later this way.